test/yaml-rest-runner/src/main/java/org/elasticsearch/test/rest/yaml/section/DoSection.java

```diff
@@ -363,7 +364,8 @@ public void execute(ClientYamlTestExecutionContext executionContext) throws IOEx
             final String testPath = executionContext.getClientYamlTestCandidate() != null
                 ? executionContext.getClientYamlTestCandidate().getTestPath()
                 : null;
-            if (executionContext.esVersion().major == Version.V_7_17_0.major && executionContext.esVersion().after(Version.V_7_17_1)) {
+
+            if (executionContext.clusterHasFeature(RestTestLegacyFeatures.REST_ELASTIC_PRODUCT_HEADER_PRESENT.id())) {
```

Comment:
Ahh it's used here.

Reply:
Yes; I made it a test-only feature because a) it's only used for tests and b) is for an older version - there would be no use for it in current prod code, but we need to support BwC tests and rolling upgrade tests.